### PR TITLE
Add and improve natspec

### LINKED ISSUE #7: Add natspec to perpetual router factory
- No natspec currently, and tough to know exactly what is going on.

_Originally posted by @mds1 in https://github.com/ScopeLift/optimizoors-perpetual-protocol/pull/1#discussion_r1177001203_

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description
- Add nat spec to the router factory contract

Closes #7

### CODE REVIEW

File: src/PerpetualRouterFactory.sol [hunk 1]
```diff
@@ -8,6 +8,9 @@ import {IAccountBalance} from "src/interface/IAccountBalance.sol";
 import {IClearingHouse} from "src/interface/IClearingHouse.sol";
 import {IVault} from "src/interface/IVault.sol";
 
+/// @notice A factory for deploying an optimized router for Perpetual protocol. A
+/// router will require an asset and wrap some piece of functionality of Perpetual
+/// protocol.
 contract PerpetualRouterFactory {
   error RouterTypeDoesNotExist();
```
[the remark this answers is not in the document]
Comment (alexkeating) [reply src/PerpetualRouterFactory.sol:15]: Is there a scopelift contract that would be a good reference for natspec best practices?

[the remark this answers is not in the document]
Comment (mds1) [reply src/PerpetualRouterFactory.sol:15]: maybe https://github.com/ScopeLift/umbra-protocol/blob/master/contracts-periphery/src/UmbraBatchSend.sol

[the remark this answers is not in the document]
Comment (mds1) [reply src/PerpetualRouterFactory.sol:15]: though it is missing some markdown formatting, e.g. types and variables should have backticks around them so `forge doc` renders it as code (it parses natspec comments as markdown and generated docs using mdbook)

[the remark this answers is not in the document]
Comment (alexkeating) [reply src/PerpetualRouterFactory.sol:15]: Thanks! This works

File: src/DepositRouter.sol
Comment (mds1) [inline src/DepositRouter.sol:1]: _(arbitrary comment location to start a thread)_

I pushed two commits:
1. First one updates comment line lengths to be 100 chars like the code, and fixes some line breaks
2. Second one changes some natspec phrasing

Please review both and make sure it looks good!

Comment (mds1) [reply src/DepositRouter.sol:1]: No worries. You can find the value in the foundry config file: https://github.com/ScopeLift/optimizoors-perpetual-protocol/blob/d445450a3e69e130b8666fd8f132e7e23a9c4798/foundry.toml#L28

Feel free to resolve this comment after reviewing my changes

**Pull request merged**